### ISSUE TITLE
Env: Arguments with spaces are causing a "Too many positional arguments" error in WP-CLI

### ISSUE THREAD

**Issue by ocean90** [WordPress/gutenberg#23320]
**Describe the bug**

Passing an argument with spaces enclosed in double quotes to WP-CLI  is currently not possible:

```
❯ npx wp-env run cli wp role create foo_bar "Foo Bar"
✖ Error while running docker-compose command.
Starting d6d348a36f1b50436296626550a1d326_mysql_1 ... done
Starting d6d348a36f1b50436296626550a1d326_wordpress_1 ... done
Error: Too many positional arguments: Bar
```

**To reproduce**
`npx wp-env run cli wp role create foo_bar "Foo Bar"`

**Additional context**
- WordPress core has the same issue: https://core.trac.wordpress.org/ticket/49491

**Comment by ocean90** [comment]
There seems to be another issue with named arguments which are not passed to the container:

```
❯ node_modules/@wordpress/env/bin/wp-env run cli wp user update 2 --display_name=Foo
✖ Error while running docker-compose command.
Starting d6d348a36f1b50436296626550a1d326_mysql_1 ... done
Starting d6d348a36f1b50436296626550a1d326_wordpress_1 ... done
Error: Need some fields to update.
```

**Comment by sainthkh** [comment]
I fixed the named arg part. But "too many positional arguments" is a bit hard to fix because the `yargs` parser parses it that way.

**Comment by sainthkh** [comment]
@ocean90 As for named arguements, it can be done with `''`: `npx wp-env run cli 'wp user update 2 --display_name=Foo'`.

**Comment by noahtallen** [comment]
How about this syntax? I think in the second case, the double quotes help wp-env understand that the arguments are for the command being passed to the CLI service, rather than arguments being passed to itself

```sh
wp-env run cli "wp role create foo_bar 'Foo Bar'"

wp-env run cli "wp user update 2 --display_name=Foo"
```

**Comment by noahtallen** [comment]
Should be fixed as of #41179, but double quotes are also useful if needed
